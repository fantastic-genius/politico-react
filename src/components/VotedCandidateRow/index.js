import React from 'react';
import logo from '@src/assets/images/apc-logo.jpg';

const VotedCandidateRow = ({candidate}) => {
  const {
    officename,
    firstname,
    lastname,
    partyname,
    logourl
  } = candidate;

  return (
    <tr>
      <td>{officename}</td>
      <td>{`${firstname} ${lastname}`}</td>
      <td><img src={logourl || logo} alt="Logo" /></td>
      <td>{partyname}</td>
    </tr>
  );
};

export default VotedCandidateRow;
