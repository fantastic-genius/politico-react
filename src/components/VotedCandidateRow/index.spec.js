import React from 'react';
import { shallow } from 'enzyme';
import VotedCandidateRow from './index';

describe('<VotedCandidateRow />', () => {
  describe('Rendering', () => {
    const candidate = {
      officename: 'President',
      firstname: 'john',
      lastname: 'mark',
      partyname: 'All peoples party',
      logourl: 'https://app.com/logo'
    }
    it('should render VotedCandidateRow correctly in debug mode', () => {
        const component = shallow(<VotedCandidateRow candidate={candidate} />);
        expect(component).toMatchSnapshot();
    });
  })

})

