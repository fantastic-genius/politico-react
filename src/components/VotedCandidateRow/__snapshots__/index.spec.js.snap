// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VotedCandidateRow /> Rendering should render VotedCandidateRow correctly in debug mode 1`] = `
<tr>
  <td>
    President
  </td>
  <td>
    john mark
  </td>
  <td>
    <img
      alt="Logo"
      src="https://app.com/logo"
    />
  </td>
  <td>
    All peoples party
  </td>
</tr>
`;
